tests/test_nb_subproc_thread.py

```diff
@@ -66,20 +80,43 @@ def test_nb_subproc_thread_stopped(tmp_path):
     th.start()
     time.sleep(2)
     assert th.is_alive()
-    th.stop()
-    th.join()
+    th.stop(wait=True)
     assert not th.is_alive()
-    # rc is None if terminated
-    assert th.returncode is None
+    # rc should be is None if terminated
+    assert th.returncode is not None
     # subprocess is terminated until it reaches kitty 4 (4 sec > 2 sec).
-    assert 'hello kitty 4' not in th.stdout
+    assert 'hello kitty 4' not in th.stderr
 
 
 def test_nb_subproc_thread_nonzero_rc(tmp_path):
-    th = NBSubprocThread(args=['cat', 'asdfasf'], on_stdout=on_stdout)
-    th.start()
-    th.join()
-    print(th.returncode, th.stdout)
-    assert th.returncode == 1
-    # check if stderr is redirected to stdout and it's captured
-    assert 'asdfasf' in th.stdout
+    for i in range(100):
```

Comment:
Why do you need the loop? Seems strange for a test.

Reply:
Previous version of Caper used a message `Queue` for streaming stdout/stderr and it was quite unstable. But it's no longer used. I removed this loop.